File: protocol/src/test/python/zfoopy/SimpleObject.py

class SimpleObject:

    c = 0  # int
    g = False  # bool

    def protocolId(self):
        return 104

    @classmethod
    def write(cls, buffer, packet):
        if packet is None:
            buffer.writeInt(0)
            return
        buffer.writeInt(-1)
        buffer.writeInt(packet.c)
        buffer.writeBool(packet.g)
        pass

    @classmethod
    def read(cls, buffer):
        length = buffer.readInt()
        if length == 0:
            return None
        beforeReadIndex = buffer.getReadOffset()
        packet = SimpleObject()
        result0 = buffer.readInt()
        packet.c = result0
        result1 = buffer.readBool() 
        packet.g = result1
        if length > 0:
            buffer.setReadOffset(beforeReadIndex + length)
        return packet

File: protocol/src/test/typescript/zfoots/packet/SimpleObject.ts
import IByteBuffer from '../IByteBuffer';
import IProtocolRegistration from '../IProtocolRegistration';


class SimpleObject {
    c: number = 0;
    g: boolean = false;
}

export class SimpleObjectRegistration implements IProtocolRegistration<SimpleObject> {
    protocolId(): number {
        return 104;
    }

    write(buffer: IByteBuffer, packet: SimpleObject | null) {
        if (packet === null) {
            buffer.writeInt(0);
            return;
        }
        buffer.writeInt(-1);
        buffer.writeInt(packet.c);
        buffer.writeBoolean(packet.g);
    }

    read(buffer: IByteBuffer): SimpleObject | null {
        const length = buffer.readInt();
        if (length === 0) {
            return null;
        }
        const beforeReadIndex = buffer.getReadOffset();
        const packet = new SimpleObject();
        const result0 = buffer.readInt();
        packet.c = result0;
        const result1 = buffer.readBoolean(); 
        packet.g = result1;
        if (length > 0) {
            buffer.setReadOffset(beforeReadIndex + length);
        }
        return packet;
    }
}

export default SimpleObject;